src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1171,36 +1176,20 @@ mobile_apis::Result::eType RegisterAppInterfaceRequest::CheckWithPolicyData() {
     }
   }
 
-  mobile_apis::Result::eType result = mobile_apis::Result::SUCCESS;
+  auto result = mobile_apis::Result::SUCCESS;
 
   // If AppHMIType is not included in policy - allow any type
-  if (!app_hmi_types.empty()) {
-    if (message[strings::msg_params].keyExists(strings::app_hmi_type)) {
-      // If AppHmiTypes are partially same, the system should allow those listed
-      // in the policy table and send warning info on missed values
-      smart_objects::SmartArray app_types =
-          *(message[strings::msg_params][strings::app_hmi_type].asArray());
-
-      std::string log;
-      CheckMissedTypes checker(app_hmi_types, log);
-      std::for_each(app_types.begin(), app_types.end(), checker);
-      if (!log.empty()) {
-        response_info_ =
-            "Following AppHmiTypes are not present in policy "
-            "table:" +
-            log;
-        result_code_ = mobile_apis::Result::WARNINGS;
-      }
+  if (!app_hmi_types_in_policy.empty()) {
+    result = ProcessingAppHMITypesPolicies(message, app_hmi_types_in_policy);
+  } else {
+    result = ProcessingAppHMITypesInMessage(message);
+    if (mobile_apis::Result::DISALLOWED == result) {
+      response_info_ =
+          "WEB_VIEW AppHmiType is absent in application policies, because they "
+          "are empty";
+      LOG4CXX_DEBUG(logger_, response_info_);
+      return result;
```

Comment:
```suggestion
  if (!app_hmi_types_in_policy.empty()) {
    result = ProcessingAppHMITypesPolicies(message, app_hmi_types_in_policy);
  }
```

Reply:
In addition I think ProcessingAppHMITypesInMessage() should be deleted.

Reply:
I need to revert this fix, Because in case if `app_hmi_types_in_policy.empty()` sdl actually do need to check ProcessingAppHMITypesInMessage for Web_view